pkg/config/expand.go

```diff
@@ -585,3 +585,45 @@ func (dg DeploymentGroup) OutputNames() []string {
 	}
 	return outputs
 }
+
+// GetIntergroupOutputNamesByGroup returns a map from group names to a list of
+// outputs that are needed by this group
+func GetIntergroupOutputNamesByGroup(thisGroup GroupName, dc DeploymentConfig) (map[GroupName][]string, error) {
```

Comment:
We can as well make it a function of `DeploymentGroup` to remove all that "index of" lines.